internal/pkg/runtime/engine/singularity/container_linux.go

```diff
@@ -2290,18 +2290,28 @@ func (c *container) prepareNetworkSetup(system *mount.System, pid int) (func(con
 		sessionNetNs = "/netns"
 	)
 
-	fakeroot := c.engine.EngineConfig.GetFakeroot()
 	net := c.engine.EngineConfig.GetNetwork()
 
 	// If we haven't requested a network namespace, or we have but with no config, we are done here
 	if !c.netNS || net == noneNet {
 		return nil, nil
 	}
 
-	// Otherwise start checking what's permitted for the current user
+	// In fakeroot mode only permit the `fakeroot` CNI config, overriding any other request.
 	euid := os.Geteuid()
+	fakeroot := c.engine.EngineConfig.GetFakeroot()
+	forceFakerootNet := false
+	if fakeroot && euid != 0 && net != fakerootNet {
+		sylog.Warningf("Only --network=%s is permitted in --fakeroot mode. You requested '%s'.", fakerootNet, net)
+		sylog.Warningf("Overriding with --network=%s", fakerootNet)
+	}
```

Comment:
Instead of duplicating 2 of the 3 checks in this if statement with the following if statement, move this code to be indented inside the following if statement and only do the 3rd check.